spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -459,6 +459,7 @@ protected void actualStart() throws Exception {
 								}
 								this.logger.error("Error creating consumer; retrying in " + nextBackOff, e);
 								doShutdown();
+								this.consumersToRestart.clear();
```

Comment:
Thanks for your feedback - I see that adding `if(started)` is more robust than the solution above. Though, if someone introduces a new location where consumers are added back to the restart list, he needs to keep in mind that an `if(started)` might be required to prevent memory leaks. 
I suppose a generally cleaner solution would be to have a fixed collection of consumers which are supposed to be running and then have everything else just update states on these instead of creating new `SimpleConsumer`.

I'll update the PR with `if(started)` checks later today .